### PR TITLE
[Ember-Migration] Implemented debounce on Inputs and Textarea of Join Section

### LINKED ISSUE #409: [Ember-Migration] Implement debounce on Inputs and Textarea
## Issue
- Currently as we are typing the data it is being also written parallely in local storage
- This is not an optimized approach to directly write the data on change to local storage

## What to do
- Implement debounce on the Inputs and Text areas to optimize the onChange behavior

### PR DESCRIPTION
**Pull request by satyam73**
### Issue

Closes #409

### What is the change?

I have added debounce in step one as of now.

### Is Development Tested?

- [x] Yes
- [ ] No

**Note :**
What to delete and what not to

- Keep the headings and answer the questions.
- Delete the examples
- Delete the description below headings `What is the change?` & `Before/After Screenshots`

### CODE REVIEW

File: app/components/join-steps/step-three.js [hunk 1]
```diff
@@ -26,12 +27,16 @@ export default class StepThreeComponent extends Component {
   }
 
   @action inputHandler(e) {
+    console.log('I am ruuning');
```
Comment (rohan09-raj) [inline app/components/join-steps/step-three.js:30]: can we please remove the `console.log` if not necessary ?

File: app/components/join-steps/step-three.js [hunk 2]
```diff
@@ -26,12 +27,16 @@ export default class StepThreeComponent extends Component {
...
-    localStorage.setItem('isValid', validated);
+    const setValToLocalStorage = () => {
+      this.data = { ...this.data, [e.target.name]: e.target.value };
+      localStorage.setItem('stepThreeData', JSON.stringify(this.data));
+      const validated =
+        validator(this.data.whyRds, 100) && validator(this.data.foundFrom, 1);
+      this.setIsValid(validated);
+      localStorage.setItem('isValid', validated);
+    };
+    debounce(this.data, setValToLocalStorage, 1000);
```
Comment (rohan09-raj) [inline app/components/join-steps/step-three.js:40]: We can maybe use a `constant` for the debounce timer

**Review by rohan09-raj — CHANGES_REQUESTED** [review]

**Review by shubhamsigdar1 — COMMENTED** [review]
Thanku @satyam73 @rohan09-raj through your issue get to know about debounce
and I reviewed all okk just go through rohan comment

File: app/components/join-steps/step-three.js (hunk 1 above)
Comment (satyam73) [reply app/components/join-steps/step-three.js]: removed✅

File: app/components/join-steps/step-three.js (hunk 2 above)
Comment (satyam73) [reply app/components/join-steps/step-three.js]: added constant✅

**Review by rohan09-raj — APPROVED** [review]

**Pull request merged**